en/skills/11. Tables.ipynb

```diff
@@ -0,0 +1,458 @@
+      },
+      "id": "8zYbmXsRlpVP"
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import csv\n",
+        "\n",
+        "# Read the existing data from the CSV file\n",
+        "with open('data.csv', newline='') as file:\n",
+        "    reader = list(csv.reader(file))\n",
+        "\n",
+        "# Add new columns and information\n",
+        "reader[0].extend([\"Spanish\"])\n",
+        "reader[1].extend([\"Lejos\"])\n",
+        "reader[2].extend([\"De nuevo\"])\n",
+        "reader[3].extend([\"Espérame\"])\n",
+        "\n",
+        "# Write the updated data back to the CSV file\n",
+        "with open('data.csv', 'w', newline='') as file:\n",
+        "    writer = csv.writer(file)\n",
+        "    writer.writerows(reader)"
+      ],
+      "metadata": {
+        "id": "BNSm7VUKl5bJ"
+      },
+      "id": "BNSm7VUKl5bJ",
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "#### Exercise 3"
+      ],
+      "metadata": {
+        "id": "MPTBYr2SlKBr"
+      },
+      "id": "MPTBYr2SlKBr"
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "###### Show answer"
+      ],
+      "metadata": {
+        "id": "_Et4LtcClOdW"
+      },
+      "id": "_Et4LtcClOdW"
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import csv\n",
+        "\n",
+        "# Read the existing data from the CSV file\n",
+        "with open('myself.csv', newline='') as file:\n",
+        "    reader = list(csv.reader(file))\n",
+        "\n",
+        "# Add new columns and information\n",
+        "reader[0].extend([\"Favorite Color\", \"Favorite Food\"])\n",
+        "reader[1].extend([\"Blue\", \"Pizza\"])\n",
+        "\n",
+        "# Write the updated data back to the CSV file\n",
+        "with open('myself.csv', 'w', newline='') as file:\n",
+        "    writer = csv.writer(file)\n",
+        "    writer.writerows(reader)"
+      ],
+      "metadata": {
+        "id": "ymg1oacIlN6o"
+      },
+      "id": "ymg1oacIlN6o",
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "##### Add more information to your myself.csv file. Create a new entry with your favorite color and favorite food. Then, update the file by adding a new column called Favorite Color and Favorite Food."
+      ],
+      "metadata": {
+        "id": "k-Yp3t7TlRfY"
+      },
+      "id": "k-Yp3t7TlRfY"
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "#TODO: Add new columns"
+      ],
+      "metadata": {
+        "id": "G1UrwjiDlZp4"
+      },
+      "id": "G1UrwjiDlZp4",
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "id": "4346b659",
+      "metadata": {
+        "id": "4346b659"
+      },
+      "source": [
+        "### Why Use Tables (CSVs) Over Dictionaries?\n",
```

Comment:
Isn't this a bit of an apples and oranges comparison? Since CSVs are a persistent file type and dictionaries are an in-memory representation?

Reply:
Yes, the logic was that people usually learn about dictionaries but not csv or vice versa, so it provides a reason. Not saying one is better than the other. Not sure if that's an issue or what do you mean exactly.